### PR TITLE
Adding delete_order function

### PR DESCRIPTION
**Pull request by enrique-ayala**

### CODE REVIEW

File: src/orders/app.py [hunk 1]
```diff
@@ -139,10 +139,7 @@ def retrieve_orders(event: Dict[str, Any], context: LambdaContext) -> Dict[str,
         if is_auth is False:
             raise AuthError("User is not allow to retrieve orders")
 
-        filter_date = doorman.get_query_param_from_request(
```
Comment (MarcoBurgos) [inline src/orders/app.py:142]: is this for delete?

File: src/orders/app.py [hunk 2]
```diff
@@ -168,3 +165,70 @@ def retrieve_orders(event: Dict[str, Any], context: LambdaContext) -> Dict[str,
...
+
+        dao = OrderDAO()
+        delete_response = dao.delete_order(
+            delivery_date=order_to_delete.delivery_date, order_id=order_to_delete.id)
+
+        if delete_response["status"] == "success":
+            logger.info(f"Order with ID {order_id} on {delivery_date} deleted")
+            return doorman.build_response(
+                payload={"message": delete_response["message"]}, 
+                status_code=200
```
Comment (MarcoBurgos) [inline src/orders/app.py]: is should be 204

Comment (enrique-ayala) [reply src/orders/app.py]: fixed

File: src/orders/app.py (hunk 1 above)
Comment (enrique-ayala) [reply src/orders/app.py:142]: This for retrieve_orders , now using a new method get_date_param_from_request() .

Comment (MarcoBurgos) [reply src/orders/app.py:142]: but what if we want to get another param, lets say a page or limit, when we include pagination in the backend

Comment (infinity-cloud-solutions) [reply src/orders/app.py:142]: In that case, you can  use doorman.get_query_param_from_request.

Comment (enrique-ayala) [reply src/orders/app.py:142]: In that case, you can use doorman.get_query_param_from_request.

Comment (enrique-ayala) [reply src/orders/app.py:142]: get_query_param_from_request is no longer coupled to a date field

**Review by MarcoBurgos — APPROVED** [review]

**Pull request closed without merging**